rotkehlchen/db/addressbook.py

```diff
@@ -109,9 +109,28 @@ def update_addressbook_entries(
             book_type: AddressbookType,
             entries: list[AddressbookEntry],
     ) -> None:
-        """Updates names of addressbook entries."""
+        """Updates names of addressbook entries.
+        If blockchain is None then make sure that the same address doesn't appear in combination
+        with other blockchain values.
+        """
         with self.write_ctx(book_type) as write_cursor:
             for entry in entries:
+                if (
+                    entry.blockchain is None and
+                    write_cursor.execute(
+                        'SELECT COUNT(*) FROM address_book WHERE address = ? AND blockchain IS NOT ?',  # noqa: E501
+                        (entry.address, ANY_BLOCKCHAIN_ADDRESSBOOK_VALUE),
+                    ).fetchone()[0] > 0
+                ):
+                    try:
+                        # Use add_addressbook_entries to delete blockchain specific entries
+                        # and replace them with a single entry.
+                        self.add_addressbook_entries(write_cursor, [entry])
+                        continue  # skip update query, we just added a new entry instead
+                    except InputError:
+                        # Single entry already exists, it must be updated using query below
+                        pass
```

Comment:
use supress instead. We use it at some places. Keep the comment to explain why this is being used

Reply:
Can you also create a test that reproduces the error that luki has and that works after your change?